server/mergin/sync/public_api_controller.py

```diff
@@ -821,8 +824,9 @@ def project_push(namespace, project_name):
     if not ws:
         abort(404)
 
-    if ws.disk_usage() + additional_disk_usage > ws.storage:
-        abort(400, "You have reached a data limit")
+    requested_storage = ws.disk_usage() + additional_disk_usage
+    if requested_storage > ws.storage:
+        abort(make_response(jsonify(StorageLimitHit(requested_storage, ws.storage).to_dict()), 422))
```

Comment:
why not return jsonify(StorageLimitHit(current_disk_usage, ws.storage).to_dict()) ?

I think we should return current usage, not requested storage